golem-worker-executor-base/src/durable_host/rdbms/mysql.rs

```diff
@@ -0,0 +1,1017 @@
+    transaction_handle: Option<u32>,
+}
+
+impl DbResultStreamEntry {
+    fn new(
+        request: RdbmsRequest<MysqlType>,
+        state: DbResultStreamState,
+        transaction_handle: Option<u32>,
+    ) -> Self {
+        Self {
+            request,
+            state,
+            transaction_handle,
+        }
+    }
+
+    fn set_open(
+        &mut self,
+        value: Arc<dyn crate::services::rdbms::DbResultStream<MysqlType> + Send + Sync>,
+    ) {
+        self.state = DbResultStreamState::Open(value);
+    }
+}
+
+#[derive(Clone)]
+pub enum DbResultStreamState {
+    New,
+    Open(Arc<dyn crate::services::rdbms::DbResultStream<MysqlType> + Send + Sync>),
+}
+
+async fn get_db_query_stream<Ctx: WorkerCtx>(
+    ctx: &mut DurableWorkerCtx<Ctx>,
+    entry: &Resource<DbResultStreamEntry>,
+) -> Result<Arc<dyn crate::services::rdbms::DbResultStream<MysqlType> + Send + Sync>, RdbmsError> {
+    let query_stream_entry = ctx
+        .as_wasi_view()
+        .table()
+        .get::<DbResultStreamEntry>(entry)
+        .map_err(|e| RdbmsError::Other(e.to_string()))?
+        .clone();
+
+    match query_stream_entry.state {
+        DbResultStreamState::New => {
+            let query_stream = match query_stream_entry.transaction_handle {
+                Some(transaction_handle) => {
+                    let (_, transaction) =
+                        get_db_transaction(ctx, &Resource::new_own(transaction_handle)).await?;
+                    transaction
+                        .query_stream(
+                            &query_stream_entry.request.statement,
+                            query_stream_entry.request.params,
+                        )
+                        .await
+                }
+                None => {
+                    ctx.state
+                        .rdbms_service
+                        .mysql()
+                        .query_stream(
+                            &query_stream_entry.request.pool_key,
+                            &ctx.state.owned_worker_id.worker_id,
+                            &query_stream_entry.request.statement,
+                            query_stream_entry.request.params,
+                        )
+                        .await
+                }
+            };
+            match query_stream {
+                Ok(query_stream) => {
+                    ctx.as_wasi_view()
+                        .table()
+                        .get_mut::<DbResultStreamEntry>(entry)
+                        .map_err(|e| RdbmsError::Other(e.to_string()))?
+                        .set_open(query_stream.clone());
+
+                    Ok(query_stream)
+                }
+                Err(e) => Err(e),
+            }
+        }
+        DbResultStreamState::Open(query_stream) => Ok(query_stream),
+    }
+}
+
+#[async_trait]
+impl<Ctx: WorkerCtx> HostDbResultStream for DurableWorkerCtx<Ctx> {
+    async fn get_columns(
+        &mut self,
+        self_: Resource<DbResultStreamEntry>,
```

Comment:
Why is this named as `self_`?

Reply:
that's the binding generator's doing, i think. It is consistent with the other host resource implementations